src/utils/types.d.ts

```diff
@@ -6,3 +6,9 @@ interface Balance {
   nonce: number | bigint;
   address: string;
 }
+
+interface Promise<T> {
+  // Improved inferred type for `.catch((err) => err)` pattern. Previously it would result in `any`.
+  // TODO: Return type should more accurately be `Promise<T | TResult>`, but existing code doesn't use type guards.
+  catch<TResult = never>(onrejected?: ((reason: Error | 'error type is unknown') => TResult | PromiseLike<TResult>) | undefined | null): Promise<T & TResult>;
```

Comment:
why is this called TResult?

wouldn't the return type be `Promise<T | TResult>` instead of `&`?

Reply:
Added a comment with original Promise.catch type `catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): Promise<T | TResult>;`. A lot of generics are named starting with `T` :shrug: 

`T | TResult` would be more accurate, but it creates a lot of type errors because typescript doesn't consider `account?.general` to be a typeguard like `'general' in account`

Reply:
ok

Reply:
Potential alternative `Promise<Partial<T> & Partial<TResult> & (T | TResult)>`